staging/src/k8s.io/client-go/util/rbuf/growing.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbuf
+
+import "errors"
+
+const (
+	defaultBufferGrowStep = 4 * 1024
+)
+
+type Growing struct {
+	data     []interface{}
+	n        int // Size of Data
+	beg      int // First available element
+	readable int // Number of data items available
+	growStep int // Size increment if buffer is full
+}
+
+func NewDefaultGrowing() *Growing {
+	return NewGrowing(defaultBufferGrowStep, defaultBufferGrowStep)
+}
+
+func NewGrowing(initialSize, growStep int) *Growing {
+	return &Growing{
+		data:     make([]interface{}, initialSize),
+		n:        initialSize,
+		growStep: growStep,
+	}
+}
+
+func (r *Growing) Readable() int {
+	return r.readable
+}
+
+func (r *Growing) ReadOne() interface{} {
+	if r.readable == 0 {
+		panic(errors.New("no elements available in the buffer"))
```

Comment:
Yes, I'll change it to return `bool`. I actually thought about it before submitting the PR but decided to get feedback first. Thanks.